src/main/java/sensorval/SensorValidator.java

```diff
@@ -2,30 +2,50 @@
 
 import java.util.List;
 
-public class SensorValidator 
-{
-    public static boolean _give_me_a_good_name(double value, double nextValue, double maxDelta) {
-        if(nextValue - value > maxDelta) {
-            return false;
-        }
-        return true;
+public class SensorValidator {
+
+    private static final double MAX_SOC_VALUE = 0.05;
+    private static final double MAX_CURRENT_VALUE = 0.1;
+
+    private SensorValidator() {
     }
-    public static boolean validateSOCreadings(List<Double> values) {
+
+    private static boolean differenceInIndexValueGreaterThanMaxThreshold(
+          double value,
+          double nextValue,
+          double maxDelta
+    )
+    {
+        return (nextValue - value > maxDelta);
+    }
+
+    private static boolean validateReadings(
+          List<Double> values,
+          double maxDeltaValue
+    )
+    {
         int lastButOneIndex = values.size() - 1;
-        for(int i = 0; i < lastButOneIndex; i++) {
-            if(!_give_me_a_good_name(values.get(i), values.get(i + 1), 0.05)) {
-            return false;
+        for (int i = 0; i < lastButOneIndex; i++) {
+            if (differenceInIndexValueGreaterThanMaxThreshold(values.get(i), values.get(i + 1), maxDeltaValue)) {
+                return false;
             }
         }
         return true;
     }
-    public static boolean validateCurrentreadings(List<Double> values) {
-        int lastButOneIndex = values.size() - 1;
-        for(int i = 0; i < lastButOneIndex; i++) {
-            if(!_give_me_a_good_name(values.get(i), values.get(i + 1), 0.1)) {
-            return false;
-            }
+
+    public static boolean validateSOCReadings(List<Double> values) {
+        throwExceptionForInvalidInput(values);
+        return validateReadings(values, MAX_SOC_VALUE);
+    }
+
+    public static boolean validateCurrentReadings(List<Double> values) {
+        throwExceptionForInvalidInput(values);
+        return validateReadings(values, MAX_CURRENT_VALUE);
+    }
+
+    private static void throwExceptionForInvalidInput(List<Double> values) {
```

Comment:
what if one of the value in List is null

Reply:
It will throw exception